### PR TITLE
docs(integrate-with-nuxt): add a guide to integrate with Nuxt.js

### PR DESCRIPTION
**Pull request by rayrutjes**
Closes: #249

### CODE REVIEW

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+mainTitle: Advanced
+layout: main.pug
+category: Advanced
+withHeadings: true
+navWeight: 3
+editable: true
+githubSource: docs/docgen/src/advanced/integrate-with-nuxt.md
+---
+
+[Nuxt.js](https://nuxtjs.org) is a framework built on top of Vue.js that helps you built applications robust applications.
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: helps you build robust applications

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+## Bootstrapping a Nuxt.js project
+
+We recommend you to use the official [vue-cli](https://vuejs.org/v2/guide/installation.html#CLI) to bootstrap your Nuxt.js project, along with the [nuxt/starter template](https://github.com/nuxt/nuxt.js/tree/dev/start).
+
+```shell
+$ npm install --global vue-cli
+$ vue init nuxt/starter nuxt-app
+```
+
+**info:** Default settings are enough, hit `Enter ⏎` at every question
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: info --> hint

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    <ais-results>
+      <template scope="{ result }">
+        <h2>{{ result.name }}</h2>
+      </template>
+    </ais-results>
+  </ais-index>
+</template>
+```
+
+Now, if you head to `http://localhost:3000/search`, you should see your search experiences.
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: search experience

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+The only issue with the current implementation, is that the results are fetched from the browser, and it would be better if they were fetched from the server side.
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: if they were already filled the first time we open this page

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+### Make search results server side rendered
+
+Nuxt.js let's you provide a handy `asyncData` method to your component so that the server can wait for your data.
+It then passes that data to the frontend by serializing it as javascript.
+
+You will want to operate the initial query to Algolia, fetch the results and render the search experience with those results.
+That way, even if the browser does not support Javascript, results still display.
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: JavaScript

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+First thing to notice is that you now [manually pass down the search store](getting-started/search-store.html#how-to-manually-create-a-search-store) to your `<ais-index>` component. 
+This allows you to have a better control over the lifecycle of the search.
+
+In the `asyncData` method, multiple things happen:
+1. you first manually start the store which will launch the first query to Algolia `searchStore.start()`
+2. you then wait for the query to be done so that you got all the results in the store with `store.waitUntilInSync()` method
+3. you return the serialized version of the store so that it can be passed to the frontend `return { serializedSearchStore: searchStore.serialize() }`
+
+Next step is making sure to expose the `searchStore` to the template.
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: The next step is exposing the `searchStore` to the template.

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+To do so, you simply need to deserialize the previously serialized store with the [`createFromSerialized` factory method](getting-started/search-store.html#create-a-search-store-from-a-previously-serialized-store).
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: To do that, you simply

File: docs/docgen/src/advanced/integrate-with-nuxt.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+```javascript
+data () {
+  return {
+    searchStore: null
+  }
+},
+```
+
+
+
```
Comment (Haroenv) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: empty lines

File: docs/docgen/src/advanced/integrate-with-nuxt.md (hunk 2 above)
Comment (rayrutjes) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: A nice, I need to do that change eventually on other pages 👍

File: docs/docgen/src/advanced/integrate-with-nuxt.md (hunk 7 above)
Comment (rayrutjes) [inline docs/docgen/src/advanced/integrate-with-nuxt.md]: I will also remove the "simply".

**Comment by rayrutjes** [comment]
All suggestions have been implemented @Haroenv . Thanks 🚀

**Pull request merged**